### ISSUE TITLE
BeforeStep and AfterStep hooks do not allow asynchronous functions - linting (es-lint) error

### ISSUE THREAD

**Issue by ansonallard** [cucumber/cucumber-js#2037]
### 👓 What did you see?

The cucumber-js api reference for BeforeStep and AfterStep indicates that one can use async functions as the function parameter to BeforeStep and AfterStep, just like Before/After and BeforeAll/AfterAll ([API reference](https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/api_reference.md)). However, when attempting to use an async function as an argument to AfterStep, I got the following typescript linting error:
`Promise returned in function argument where a void return was expected`

After doing some poking around, of the three hook types, the `TestStepHookFunction` type only allows for a return type of void, whereas the other two types, `TestStepHookFunction` and `TestStepFunction` have the following return types: `any | Promise<any>` ([code](https://github.com/cucumber/cucumber-js/blob/3928c673378ccabf2138b7c55873f6981e9eb134/src/support_code_library_builder/types.ts#L35)).

### ✅ What did you expect to see?

I expected that `TestStepHookFunction`, `TestStepHookFunction`, and `TestStepFunction` all had the return type of `any | Promise<any>`.

### 📦 Which tool/library version are you using?

Cucumber-js version: 7.3.2
Typescript version: 4.6.3
eslint: 8.12.0

### 🔬 How could we reproduce it?

1. Install cucumber-js version 7.3.2
2. Install and apply eslint version 8.12.0
3. Create a BeforeStep and pass in an async function, something like:
```
BeforeStep(async function(): Promise<void> {
await fetch("http://github.com");
})
```
4. Linter/compiler throws expected error: `Promise returned in function argument where a void return was expected`

### 📚 Any additional context?

----
 
*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/bug_report.md)*

**Comment by aurelien-reeves** [comment]
Hello @ansonallard :)

The api reference applies to latest version of cucumber-js: >=8.0

Could you please make a try with the latest version?

**Comment by ansonallard** [comment]
Hey @aurelien-reeves,

I upgraded to cucumber js version 8.2.0 and was still experiencing the same issue.

Looks like @davidjgoss found the issue with the return types.

**Comment by davidjgoss** [comment]
This fix has been released in 8.2.1 https://github.com/cucumber/cucumber-js/releases/tag/v8.2.1
